src/moneymoney/calculateBalances.ts

```diff
@@ -4,40 +4,43 @@ import { formatDateKey } from '../lib';
 export default function calculateBalances(
   transactions: Transaction[],
   defaultCategoryIds: string[],
+  ignorePendingTransactions: boolean
 ): Balances {
   return transactions.reduce((memo, transaction) => {
-    const key = formatDateKey(transaction.bookingDate);
-    let balance = memo[key];
-    if (!balance) {
-      balance = {
-        total: 0,
-        categories: {},
-        uncategorised: {
-          amount: 0,
-          transactions: [],
-        },
-      };
-      memo[key] = balance;
-    }
+    if (transaction.booked || !ignorePendingTransactions) {
```

Comment:
style feedback: I'd convert this to an early return. 

```ts
if (ignorePendingTransactions && !transaction.booked) {
  return memo;
}
```

Rest is same right?